### PR TITLE
Validate integration parameters display name

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30343

## Description
Integration parameters display name should be capitalized and spaced using whitespace and not underscores.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 1]
```diff
@@ -883,6 +884,22 @@ def is_there_a_runnable(self) -> bool:
...
+        configuration = self.current_file.get('configuration', {})
+        parameters_display_name = [param.get('display') for param in configuration if param.get('display')]
+
+        invalid_display_names = []
+        for parameter in parameters_display_name:
+            invalid_display_names.append(parameter) if parameter and not parameter[0].isupper() or '_' in parameter \
+                else None
+
+        if invalid_display_names:
+            error, code = Errors.invalid_integration_parameters_display_name(invalid_display_names)
```
Comment (bakatzir) [inline demisto_sdk/commands/common/hook_validations/integration.py:897]: how will the error look?
conside join with `, ` the invalid_display_names

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 2]
```diff
@@ -883,6 +884,22 @@ def is_there_a_runnable(self) -> bool:
             return False
         return True
 
+    def is_valid_parameters_display_name(self) -> bool:
```
Comment (bakatzir) [inline demisto_sdk/commands/common/hook_validations/integration.py:887]: docstring with good and bad examples will be nice.

File: demisto_sdk/commands/common/tests/integration_test.py [hunk 3]
```diff
@@ -530,6 +530,26 @@ def test_is_valid_deprecated_integration(self, current, answer):
         validator.current_file = current
         assert validator.is_valid_as_deprecated() is answer
 
+    def test_valid_integration_parameters_display_name(self, integration):
```
Comment (bakatzir) [inline demisto_sdk/commands/common/tests/integration_test.py:533]: add given/when/then

File: demisto_sdk/commands/common/tests/integration_test.py [hunk 4]
```diff
@@ -530,6 +530,26 @@ def test_is_valid_deprecated_integration(self, current, answer):
...
+        integration.yml.write_dict({'configuration': [
+            {'display': 'Token'},
+            {'display': 'Username'}
+        ]})
+        structure_validator = StructureValidator(integration.yml.path, predefined_scheme='integration')
+        validator = IntegrationValidator(structure_validator)
+
+        assert validator.is_valid_parameters_display_name()
+
+    def test_invalid_integration_parameters_display_name(self, integration):
```
Comment (bakatzir) [inline demisto_sdk/commands/common/tests/integration_test.py:543]: add given/when/then

File: demisto_sdk/commands/common/hook_validations/integration.py (hunk 2 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/integration.py:887]: done

File: demisto_sdk/commands/common/tests/integration_test.py (hunk 3 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/tests/integration_test.py:533]: done

File: demisto_sdk/commands/common/tests/integration_test.py (hunk 4 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/tests/integration_test.py:543]: done

File: demisto_sdk/commands/common/hook_validations/integration.py (hunk 1 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/integration.py:897]: It looks like a list of problematic words -
The integration display names: `[username, Client_id]` are invalid, Integration parameters display name should be capitalized and spaced using whitespace and not underscores."

**Review by bakatzir — APPROVED** [review]

**Comment by ShahafBenYakir** [comment]
Add to changelog

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,7 @@
 * Added support for the incident fields in outgoing mappers in the ID set.
 * Added a validation that the taskid field and the id field under the task field are both from uuid format and contain the same value.
 * Updated the **format** command to generate uuid value for the taskid field and for the id under the task field in case they hold an invalid values.
+* Added a validation that the integration parameters display name is capitalized and spaced using whitespace and not underscores.
```
Comment (ShahafBenYakir) [inline CHANGELOG.md:8]:
```suggestion
* Added validation for integration parameter display names are capitalized and spaced using whitespace and not underscores.
```

**Review by ShahafBenYakir — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (ChanochShayner) [reply CHANGELOG.md]: done

File: demisto_sdk/commands/common/errors.py [hunk 6]
```diff
@@ -448,6 +449,13 @@ def integration_non_existent_classifier(integration_classifier):
     def integration_non_existent_mapper(integration_mapper):
         return f"The integration has a mapper {integration_mapper} which does not exist."
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_integration_parameters_display_name(invalid_display_names):
+        return f"The integration display names: {invalid_display_names} are invalid, " \
+               "Integration parameters display name should be capitalized and spaced using whitespace " \
+               "and not underscores."
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:457]:
```suggestion
               "and not underscores ( _ )."
```

Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:457]:
```suggestion
               "and not underscores ( _ )."
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,7 @@
 * Added support for the incident fields in outgoing mappers in the ID set.
 * Added a validation that the taskid field and the id field under the task field are both from uuid format and contain the same value.
 * Updated the **format** command to generate uuid value for the taskid field and for the id under the task field in case they hold an invalid values.
+* Added validation that integration parameter display names are capitalized and spaced using whitespace and not underscores.
```
Comment (roysagi) [inline CHANGELOG.md:8]:
```suggestion
* Added validation that integration parameter's display names are capitalized and spaced using whitespaces and not underscores.
```

File: demisto_sdk/commands/common/errors.py [hunk 8]
```diff
@@ -448,6 +449,13 @@ def integration_non_existent_classifier(integration_classifier):
     def integration_non_existent_mapper(integration_mapper):
         return f"The integration has a mapper {integration_mapper} which does not exist."
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_integration_parameters_display_name(invalid_display_names):
+        return f"The integration display names: {invalid_display_names} are invalid, " \
+               "Integration parameters display name should be capitalized and spaced using whitespace " \
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:456]:
```suggestion
               "Integration parameters display name should be capitalized and spaced using whitespaces " \
```

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 9]
```diff
@@ -883,6 +884,28 @@ def is_there_a_runnable(self) -> bool:
...
+        configuration = self.current_file.get('configuration', {})
+        parameters_display_name = [param.get('display') for param in configuration if param.get('display')]
+
+        invalid_display_names = []
+        for parameter in parameters_display_name:
+            invalid_display_names.append(parameter) if parameter and not parameter[0].isupper() or '_' in parameter \
+                else None
+
+        if invalid_display_names:
+            error, code = Errors.invalid_integration_parameters_display_name(invalid_display_names)
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/integration.py:903]:
```suggestion
            error_message, error_code = Errors.invalid_integration_parameters_display_name(invalid_display_names)
```
in accordance with validate's conventions

**Review by roysagi — APPROVED** [review]
Looks good, see my minor changes please.

File: demisto_sdk/commands/common/errors.py (hunk 6 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/errors.py:457]: done

Comment (ChanochShayner) [reply demisto_sdk/commands/common/errors.py:457]: duplicate.. done

File: CHANGELOG.md (hunk 7 above)
Comment (ChanochShayner) [reply CHANGELOG.md:8]: done

File: demisto_sdk/commands/common/errors.py (hunk 8 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/errors.py:456]: done

File: demisto_sdk/commands/common/hook_validations/integration.py (hunk 9 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/integration.py]: done

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**